circuit_knitting/cutting/cutting_decomposition.py

```diff
@@ -269,3 +270,71 @@ def decompose_observables(
     }
 
     return subobservables_by_subsystem
+
+
+def find_gate_cuts(
+    circuit: QuantumCircuit, num_cuts: int, transpilation_options: dict
+) -> tuple[QuantumCircuit, list[QPDBasis], list[int]]:
+    r"""
+    Find an optimized set of gates to cut, given a transpilation context.
+
+    This function seeks to reduce the depth of the transpiled
+    sub-experiments by cutting gates which result in the highest swap overhead.
+
+    Args:
+        circuit: The circuit to cut
+        num_cuts: The number of cuts to make
+        transpilation_options: A dictionary of kwargs to be passed to the Qiskit
+            ``transpile`` function.
+
+    Returns:
+        A tuple containing:
+            - A copy of the input circuit with SWAP-costly gates replaced with :class:`TwoQubitGate`\ s
+            - A list of :class:`QPDBasis` instances -- one for each QPD gate in the circuit
+            - A list of indices where the new :class:`TwoQubitGate`\ s are located in the output circuit
+    """
+    circ_copy = circuit.copy()
+
+    # Sweep the circuit num_cuts times. In each sweep, find the gate that results in the biggest
+    # reduction in depth, and replace it with a local, placeholder gate.
+    cut_indices = []
+    for cuts in range(num_cuts):
+        cut_scores = _evaluate_cuts(circ_copy, transpilation_options)
+        best_idx = cut_scores[0][0]
+        cut_indices.append(best_idx)
+        # Put a single qubit placeholder in place of the optimal gate
+        qubit0 = circ_copy.find_bit(circ_copy.data[best_idx].qubits[0]).index
+        # Use XGate as a placeholder since it will transpile to all our backends.
```

Comment:
This could affect the transpilation. Need to do a delay or some other kind of no-op